test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -160,6 +179,353 @@ public void testLogReplicationFSMTransitions() throws Exception {
+
+        // Default sync value is null so current syncType should not be set and default to SNAPSHOT.
+        Assert.assertFalse(currentReplicationInfo.hasField(ReplicationInfo.getDescriptor().findFieldByName("syncType")));
+
+        // Current SyncStatus for ReplicationInfo and SnapshotSyncInfo should be NOT_STARTED
+        Assert.assertEquals(SyncStatus.NOT_STARTED, currentReplicationVal.getSourceStatus().getReplicationInfo().getStatus());
+        Assert.assertEquals(SyncStatus.NOT_STARTED, currentReplicationVal.getSourceStatus().getReplicationInfo().getSnapshotSyncInfo().getStatus());
+
+        // Transition #1: SNAPSHOT Sync Start
+        transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+
+        // Await for update to status table from entry of InSnapshotSyncState
+        statusTableLatch.await();
+        corfuStore.unsubscribeListener(streamListener);
+
+        Set<SyncType> expectedSyncTypes = new HashSet<>(Collections.singletonList(SyncType.SNAPSHOT));
+        Set<SyncType> actualSyncTypes = new HashSet<>();
+        SyncStatus actualSyncStatus;
+        SyncStatus actualSnapshotInfoSyncStatus;
+
+        ArrayList<CorfuStreamEntries> streamEntries = streamListener.getEntries();
+        ReplicationStatus replicationStatusVal = (ReplicationStatus)
+                streamEntries.stream().findAny().get().getEntries().values().stream().findAny().get().get(0).getPayload();
+        actualSyncTypes.add(replicationStatusVal.getSourceStatus().getReplicationInfo().getSyncType());
+        actualSyncStatus = replicationStatusVal.getSourceStatus().getReplicationInfo().getStatus();
+        actualSnapshotInfoSyncStatus = replicationStatusVal.getSourceStatus().getReplicationInfo().getSnapshotSyncInfo().getStatus();
+
+        // replicationSyncStatus (outer) starts as NOT_STARTED by initializeReplicationStatusTable,
+        // and then gets updated to ONGOING in entry of InSnapshotSyncState
+        SyncStatus replicationSyncStatusAfterEntry = SyncStatus.ONGOING;
+        // snapshotInfoSyncStatus (inner) starts as NOT_STARTED by initializeReplicationStatusTable,
+        // and then gets updated to ONGOING in entry of InSnapshotSyncState
+        SyncStatus snapshotInfoSyncStatusAfterEntry = SyncStatus.ONGOING;
+
+        Assert.assertEquals(expectedSyncTypes, actualSyncTypes);
+        Assert.assertEquals(replicationSyncStatusAfterEntry, actualSyncStatus);
+        Assert.assertEquals(snapshotInfoSyncStatusAfterEntry, actualSnapshotInfoSyncStatus);
+    }
+
+    /**
+     * Verify the lastSyncType flag is being initialized properly and SnapshotSyncInfo
+     * reflects accurate sync type for log entry sync.
+     *
+     */
+    @Test
+    public void testSyncStatusUpdatesForLogEntryOnInit() throws Exception {
+        final int updateToStatusTableFromOnEntry = 1;
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
```

Comment:
initLogReplicationFSM() creates an incoming session for which the Source status is not initialized - 
https://github.com/CorfuDB/CorfuDB/blob/log_replication_v2/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java#L137

In the test, the purpose is to test for the SourceStatus.  You can either write a separate initialize method or extend the current one to take an 'incoming' param, whichever is cleaner.

Additionally, the verification and comment on L278 will change because the syncType field is present and has been initialized to NOT_STARTED.  I think the verification on L282 is working now even though those fields are not initialized because they are the default enum values in v2.  

Similar test change will be needed for other tests.